go/tools/gopackagesdriver/bazel_json_builder.go

```diff
@@ -125,12 +125,12 @@ func (b *BazelJSONBuilder) queryFromRequests(requests ...string) string {
 		if strings.HasSuffix(request, ".go") {
 			f := strings.TrimPrefix(request, "file=")
 			result = b.fileQuery(f)
+		} else if bazelQueryScope != "" {
+			result = b.packageQuery(request)
```

Comment:
Just to understand this fix better: Would it make sense to restrict the `localQuery` to the `packageQuery` (e.g. via `intersect`) instead of always using the `packageQuery`?

Reply:
Hi, the packageQuery currently returns a bazel query formatted to find dependencies by import path (line 116, `kind("%s", attr(importpath, "%s", deps(%s)))`), so even if localQuery was restricted to the bazelQueryScope, it would not produce the same required output. It seems this behavior change happened after commit b84cd75919a40ce7d1409a6effbdd9a343cdaf72.

Reply:
just an FYI but I believe this PR may have introduced a bit of a subtle bug when gopls sends "./..." and "bultin" args to the driver: https://github.com/bazelbuild/rules_go/issues/3999